### PR TITLE
QA-752 - Added 4 new endpoints

### PR DESCRIPTION
**Pull request by bw-lee**
***Changes***
Added 4 new endpoints
```
GET /projects/#{project_id}/signals/groups/#{group_id}/signals
GET /brandwatch/usageReport/client/#{clientId}/start/#{startDate}/end/#{endDate}
GET /projects/#{project_id}/query/#{query_id}/mentionfind
POST /brandwatch/log-level
```

***Rubocop***
111 files inspected, no offenses detected

***Rspec***
Finished in 0.02681 seconds (files took 1.68 seconds to load)
41 examples, 0 failures
[Coveralls] Outside the Travis environment, not sending data.
Coverage report generated for RSpec to /Users/lee/brandwatch/bwapi/coverage. 987 / 1354 LOC (72.9%) covered.

### CODE REVIEW

File: lib/bwapi/client/brandwatch/log_level.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module BWAPI
+  class Client
+    module Brandwatch
+      # Loglevel module for brandwatch/log-level endpoint
+      module LogLevel
+        # Change the API LogLevel
+        #
+        # @param opts [Hash] options hash of parameters
+        # @option opts [string] log_level for API2
```
Comment (benSlaughter) [inline lib/bwapi/client/brandwatch/log_level.rb]: can you change the option description to match the api param of 'logLevel'

Comment (benSlaughter) [inline lib/bwapi/client/brandwatch/log_level.rb]: Missing `# @return [Hash] ...` comment

File: lib/bwapi/client/brandwatch/usage_report.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  class Client
+    module Brandwatch
+      # usageReport module for brandwatch/usageReport endpoint
+      module UsageReport
+        # Get the Clients Mention Usage report
+        #
+        # @param clientId [Integer] Id of client
+        # @param startDate [DateTime] Report startDate
+        # @param endDate [DateTime] Report endDate
+        # @param opts [Hash] options hash of parameters
```
Comment (benSlaughter) [inline lib/bwapi/client/brandwatch/usage_report.rb]: Missing `# @return [Hash] ...` comment

File: lib/bwapi/client/projects/query/mentionfind.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    module Projects
+      module Query
+        # mentionfind module for projects/project_id/query/query_id/mentionfind endpoint
+        module Mentionfind
+          # Returns the mention for a specific URL
+          #
+          # @param project_id [Integer] Id of project
+          # @param query id [Integer] Id of the query
+          # @param opts [hash] options Hash of parameters
+          # @option opts [string] URL of the mention
```
Comment (benSlaughter) [inline lib/bwapi/client/projects/query/mentionfind.rb]: Missing `# @return [Hash] ...` comment

**Comment by benSlaughter** [comment]
Few small comments, but other than that the endpoints are solid

File: lib/bwapi/client/brandwatch/usage_report.rb (hunk 2 above)
Comment (bw-lee) [inline lib/bwapi/client/brandwatch/usage_report.rb]: Added

File: lib/bwapi/client/projects/query/mentionfind.rb (hunk 3 above)
Comment (bw-lee) [inline lib/bwapi/client/projects/query/mentionfind.rb]: added

**Comment by benSlaughter** [comment]
Looks good, once the build comes back I'll merge

File: lib/bwapi/client/projects/query/mentionfind.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module BWAPI
+  class Client
+    module Projects
+      module Query
+        # mentionfind module for projects/project_id/query/query_id/mentionfind endpoint
+        module Mentionfind
```
Comment (jonathanchrisp) [inline lib/bwapi/client/projects/query/mentionfind.rb]: I think this should be `MentionFind`

**Comment by jonathanchrisp** [comment]
One minor comment added, not major but would be good to investigate.

**Comment by bw-lee** [comment]
All updates complete. @benSlaughter can we merge this as we have the 3 endpoints ready to go in API project.  cheers

**Pull request merged**